compiler/rustc_data_structures/src/vec_map.rs

```diff
@@ -33,6 +33,13 @@ where
         self.0.iter().find(|(key, _)| k == key.borrow()).map(|elem| &elem.1)
     }
 
+    pub fn get_by<P>(&self, predicate: P) -> Option<&V>
```

Comment:
doc comment :)

Reply:
```suggestion
    pub fn get_by<P>(&self, predicate: impl FnMut(&(K,V)) -> Option<&V>
```

Reply:
I'm tempted to say it should be `FnMut(&K)`, even

Reply:
although that's clearly less general

Reply:
Done